src/daemon/rpmostree-sysroot-upgrader.cxx

```diff
@@ -1439,9 +1440,19 @@ rpmostree_sysroot_upgrader_deploy (RpmOstreeSysrootUpgrader *self,
   if (g_hash_table_size (rpmostree_origin_get_initramfs_etc_files (self->origin)) > 0)
     {
       glnx_fd_close int fd = -1;
-      if (!ror_initramfs_overlay_generate (rpmostree_origin_get_initramfs_etc_files (self->origin),
-                                           &fd, cancellable, error))
-        return glnx_prefix_error (error, "Generating initramfs overlay");
+      auto etc_files_gset = rpmostree_origin_get_initramfs_etc_files (self->origin);
+      rust::Vec<rust::String> etc_files;
+      GLNX_HASH_TABLE_FOREACH(etc_files_gset, const char *, key)
+        {
+          etc_files.push_back(std::string(key));
+        }
+      try {
+        fd = rpmostreecxx::initramfs_overlay_generate (etc_files, *cancellable);
+      } catch (std::exception& e) {
+        std::ostringstream msg;
+        msg << "Generating initramfs overlay: " << e.what();
+        throw std::runtime_error(msg.str());
```

Comment:
This is some rather ugly code to do the C++ exception equivalent of `return glnx_prefix_error()`.  I personally really dislike C++ `<<` for formatting.  It looks like there's https://en.cppreference.com/w/cpp/utility/format/format in C++20 but that might be too new.  Maybe the simple cases are covered with a `utils::rethrow_prefixed (e, "Generating initramfs overlay");`.
Will add that if we do more of this.

Reply:
OK ended up adding a helper for this.

Reply:
Thanks for the pre-emptive helper, I'd expect this pattern to become very common through the codebase.